### ISSUE TITLE
to_json()   ?-character

### ISSUE THREAD

**Issue by AlexHolly** [godotengine/godot#3283]
```
{
   "payload":"?"
}.to_json()
```
gives me
```
"
{
   "payload":"\?"
}
"
```

My Server(java) is not able to parse the \ is this a bug?

I also tested with https://jsonformatter.curiousconcept.com/ and it tells me that this is not a valid json.
```
{
   "payload":"\?"
}
```

**Comment by bojidar-bg** [comment]
This was worked around in #2521

**Comment by AlexHolly** [comment]
I don't understand what the workaround does but if the godot json parser now accepts \? it is not correct.

I made the same test in java it also creates the json without the \

**Comment by bojidar-bg** [comment]
Actually the main problem is that the escape function used for JSON escapes some characters that it shouldn't.
Tracing the problem (by hand), shows:
1. [Dictionary::to_json](https://github.com/godotengine/godot/blob/master/core/dictionary.cpp#L215)
2. [JSON::print](https://github.com/godotengine/godot/blob/master/core/io/json.cpp#L97) -> [JSON::print_var](https://github.com/godotengine/godot/blob/master/core/io/json.cpp#L47)
3. [JSON::print_var@default case](https://github.com/godotengine/godot/blob/master/core/io/json.cpp#L89)
4. [String::c_escape](https://github.com/godotengine/godot/blob/master/core/ustring.cpp#L3156)

The last function escapes the following characters:

Source | Conversion | Comment
------ | ---------- | -------
`\` | `\\` | Same as in JSON
Audible bell character | `\a` | **Not supported in JSON**
Backspace character | `\b` | Same as in JSON
Form feed - new page | `\f` | Same as in JSON
New line | `\n` | Same an in JSON
Carriage return | `\r` | Same as in JSON
Tab character | `\t` | Same as in JSON
Vertical tab character | `\v` | **Not supported in JSON**
`'` | `\'` | Same as in JSON
`"` | `\"` | Same as in JSON
`?` | `\?` | **Not supported in JSON**

As you can see, the only problematic characters are `\a`, `\v` and `?`. Actually, the first two are not problematic, as they would occur very rarely, if ever, so the only problem is the question mark.

**Comment by neikeq** [comment]
I think this was fixed by 4fdab4f
